xtable-api/src/main/java/org/apache/xtable/spi/sync/CatalogSync.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+ 
+package org.apache.xtable.spi.sync;
+
+import static org.apache.xtable.spi.sync.CatalogUtils.hasStorageDescriptorLocationChanged;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+
+import org.apache.commons.lang3.StringUtils;
+
+import org.apache.xtable.conversion.ExternalCatalog;
+import org.apache.xtable.model.InternalTable;
+import org.apache.xtable.model.sync.SyncResult;
+import org.apache.xtable.model.sync.SyncResult.CatalogSyncStatus;
+
+@Log4j2
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class CatalogSync {
+  private static final CatalogSync INSTANCE = new CatalogSync();
+
+  public static CatalogSync getInstance() {
+    return INSTANCE;
+  }
+
+  public <TABLE> Map<String, List<CatalogSyncStatus>> syncTable(
+      Collection<CatalogSyncClient<TABLE>> catalogSyncClients, InternalTable table) {
+    Map<String, List<CatalogSyncStatus>> results = new HashMap<>();
+    for (CatalogSyncClient<TABLE> catalogSyncClient : catalogSyncClients) {
+      results.computeIfAbsent(catalogSyncClient.getTableFormat(), k -> new ArrayList<>());
```

Comment:
Simplify the below by simply assigning a list here that you can add to instead of making get calls below

Reply:
Are you suggesting we add them to list of List<Pair<TableFormat, CatalogSyncStatus>> and then a collect.toMap ? Using the get calls to avoid the list and collectors.toMap.

Reply:
Simplified this.